eng/versioning/external_dependencies.txt

```diff
@@ -243,7 +243,7 @@ org.junit.vintage:junit-vintage-engine;5.9.1
 org.openjdk.jmh:jmh-core;1.22
 org.openjdk.jmh:jmh-generator-annprocess;1.22
 org.spockframework:spock-core;2.3-groovy-4.0
-org.testng:testng;7.5
+org.testng:testng;7.7.1
```

Comment:
There is an on-going discussion around this upgrade as TestNG 7.6 and later requires Java 11 as a baseline

Reply:
Ok that makes sense.